LibreNMS/Util/Rewrite.php

```diff
@@ -155,9 +155,13 @@ public static function readableMac($mac)
      */
     public static function readableOUI($mac)
     {
-        $key = 'OUIDB-' . (substr($mac, 0, 6));
+        $cached = Cache::get('OUIDB-' . (substr($mac, 0, 6)), '');
+        if ($cached == 'IEEE Registration Authority') {
+            // Then we may have a shorter prefix, so let's try them one ater the other, ordered by probability
+            return Cache::get('OUIDB-' . (substr($mac, 0, 7)), Cache::get('OUIDB-' . (substr($mac, 0, 9))));
```

Comment:
This indicates to me that the cache key is wrong.

Reply:
I would try the longer one first, prevents collision

Reply:
Also, this fetches both, determines if the /28 one is empty, then returns the /36 one.

Reply:
> I would try the longer one first, prevents collision

In fact, the OUI is always defined. If OUI is 'IEEE Registration Authority' then it means a small range exists. It can be either all /28 or all /36. Never /36 in /28. And statistically, it is /28 most of the time, only a very limited number is /36. That's why I ordered it that way.

Reply:
> This indicates to me that the cache key is wrong.

Could you be a little bit more precise ? I don't get that one.

Reply:
> Also, this fetches both, determines if the /28 one is empty, then returns the /36 one.

As I have no way to know which one I search in advance, I can only fetch one and the other. Just trying to fetch the most probable one first.